core/src/main/java/com/linecorp/armeria/server/HttpHeaderUtil.java

```diff
@@ -141,32 +147,54 @@ private static InetAddress createInetAddress(String address) throws UnknownHostE
         // Remote quotes. e.g. "[2001:db8:cafe::17]:4711" => [2001:db8:cafe::17]:4711
         final String addr = firstChar == '"' ? QUOTED_STRING_TRIMMER.trimFrom(address) : address;
 
-        if (NetUtil.isValidIpV4Address(addr) || NetUtil.isValidIpV6Address(addr)) {
-            return InetAddress.getByAddress(NetUtil.createByteArrayFromIpAddressString(addr));
+        final InetSocketAddress inetSocketAddress;
+        try {
+            if (addr.charAt(0) == '[') {
+                inetSocketAddress = createInetSocketAddress(addr, ']');
+            } else {
+                inetSocketAddress = createInetSocketAddress(addr, ':');
+            }
+            // Log debug message if it is an invalid address.
+            if (inetSocketAddress == null) {
```

Comment:
This never happens, right?

Reply:
Yes, it's my mistake. Thanks for pointing this out!